src/node.cpp

```diff
@@ -280,13 +284,16 @@ Node* Node::unserialize(MegaClient* client, string* d, node_vector* dp)
         fa = NULL;
     }
 
-    for (i = 8; i--;)
+//    for (i = 8; i--;)
+    for (i = 7; i--;)
     {
         if (ptr + (unsigned char)*ptr < end)
         {
             ptr += (unsigned char)*ptr + 1;
         }
     }
+    version_flag = MemAccess::get<char>(ptr);
```

Comment:
Maybe it would be better to put this on the first free byte, instead of in the last one.
That way, the other bytes could be used to store optional data depending on the version